### PR TITLE
Enabling PWM support for STM32 L152RE Platform

### PR DESCRIPTION
**Pull request by sarthsmart**
Enabled PWM support for STM32 L152RE target platform.

Verified PWM signals on timer3 using test application `tests/drivers/pwm`

### CODE REVIEW

File: dts/arm/st/l1/stm32l1.dtsi [hunk 1]
```diff
@@ -32,6 +32,30 @@
...
+
+				write-block-size = <4>;
+			};
+		};
+
+		rcc: rcc@40023800 {
+			compatible = "st,stm32-rcc";
+			#clock-cells = <2>;
+			reg = <0x40023800 0x400>;
+		};
```
Comment (gmarull) [inline dts/arm/st/l1/stm32l1.dtsi:57]: rcc/flash changes should go into its own commit. is this change made because others define flash/rcc at the top?

**Review by gmarull — CHANGES_REQUESTED** [review]

File: dts/arm/st/l1/stm32l1.dtsi (hunk 1 above)
Comment (sarthsmart) [reply dts/arm/st/l1/stm32l1.dtsi:57]: @gmarull Yes. It is moved from the bottom to top.

Comment (FRASTM) [reply dts/arm/st/l1/stm32l1.dtsi:57]: yes, this is disturbing to see a change on this part of the dts, could this be avoided ?

Comment (sarthsmart) [reply dts/arm/st/l1/stm32l1.dtsi:57]: @FRASTM This specific change was made to add nodes after rcc/flash, looks similar to other platform files.

Comment (gmarull) [reply dts/arm/st/l1/stm32l1.dtsi:57]: @sarthsmart can you move these changes to a separate commit?

**Comment by FRASTM** [comment]
The compilation error in the drivers/pwm/pwm_stm32.c :
It seems the stm32L1 soc series does not support the break function on any of its timer, so should the pwm_stm32.c include a flag on **CONFIG_SOC_SERIES_STM32L1X** in the pwm_stm32_init function:
```
#ifndef CONFIG_SOC_SERIES_STM32L0X
	/* enable outputs and counter */
	if (IS_TIM_BREAK_INSTANCE(cfg->timer)) {
		LL_TIM_EnableAllOutputs(cfg->timer);
	}
#endif
```

File: dts/arm/st/l1/stm32l1.dtsi (hunk 1 above)
Comment (FRASTM) [reply dts/arm/st/l1/stm32l1.dtsi:57]: +1

Comment (sarthsmart) [reply dts/arm/st/l1/stm32l1.dtsi:57]: @FRASTM @erwango @gmarull Separate commit for changing the sequence of rcc/flash has been submitted.

**Review by gmarull — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

File: dts/arm/st/l1/stm32l1.dtsi (hunk 1 above)
Comment (FRASTM) [reply dts/arm/st/l1/stm32l1.dtsi:57]: thanks, this PR could now be closed
